examples/generative/finetune_stable_diffusion.py

```diff
@@ -0,0 +1,507 @@
+from keras_cv.models.stable_diffusion.diffusion_model import DiffusionModel
+from keras_cv.models.stable_diffusion.image_encoder import ImageEncoder
+from keras_cv.models.stable_diffusion.noise_scheduler import NoiseScheduler
+from keras_cv.models.stable_diffusion.text_encoder import TextEncoder
+from tensorflow import keras
+
+"""
+## Data loading
+
+The dataset is
+[Pokémon BLIP captions](https://huggingface.co/datasets/lambdalabs/pokemon-blip-captions).
+However, we'll use a slightly different version which was derived from the original
+dataset to fit better with `tf.data`. Refer to
+[the documentation](https://huggingface.co/datasets/sayakpaul/pokemon-blip-original-version)
+for more details.
+"""
+
+data_path = tf.keras.utils.get_file(
+    origin="https://huggingface.co/datasets/sayakpaul/pokemon-blip-original-version/resolve/main/pokemon_dataset.tar.gz",
+    untar=True,
+)
+
+data_frame = pd.read_csv(os.path.join(data_path, "data.csv"))
+
+data_frame["image_path"] = data_frame["image_path"].apply(
+    lambda x: os.path.join(data_path, x)
+)
+data_frame.head()
+
+"""
+Since we have only 833 {image,caption} pairs we can precompute the text embeddings from
+the captions. Moreover, the text encoder will be kept frozen during the course of
+fine-tuning. So, we can save some compute by doing this.
+
+Before we employ the text encoder, we need to tokenize the captions.
+"""
+
+PADDING_TOKEN = 49407
+MAX_PROMPT_LENGTH = 77
+
+# Load the tokenizer.
+tokenizer = SimpleTokenizer()
+
+#  Method to tokenize and pad the tokens.
+def process_text(caption):
+    tokens = tokenizer.encode(caption)
+    tokens = tokens + [PADDING_TOKEN] * (MAX_PROMPT_LENGTH - len(tokens))
+    return np.array(tokens)
+
+
+# Collate the tokenized captions into an array.
+tokenized_texts = np.empty((len(data_frame), MAX_PROMPT_LENGTH))
+
+all_captions = list(data_frame["caption"].values)
+for i, caption in enumerate(all_captions):
+    tokenized_texts[i] = process_text(caption)
+
+"""
+## Prepare a `tf.data.Dataset`
+
+In this section, we'll prepare a `tf.data.Dataset` object from the input image file paths
+and their corresponding caption tokens. The section will include the following:
+
+* Pre-computation of the text embeddings from the tokenized captions.
+* Loading and augmentation of the input images.
+* Shuffling and batching of the dataset.
+"""
+
+RESOLUTION = 256
+AUTO = tf.data.AUTOTUNE
+POS_IDS = tf.convert_to_tensor([list(range(MAX_PROMPT_LENGTH))], dtype=tf.int32)
+
+augmenter = keras_cv.layers.Augmenter(
+    layers=[
+        keras_cv.layers.CenterCrop(RESOLUTION, RESOLUTION),
+        keras_cv.layers.RandomFlip(),
+        tf.keras.layers.Rescaling(scale=1.0 / 127.5, offset=-1),
+    ]
+)
+text_encoder = TextEncoder(MAX_PROMPT_LENGTH)
+
+
+def process_image(image_path, tokenized_text):
+    image = tf.io.read_file(image_path)
+    image = tf.io.decode_png(image, 3)
+    image = tf.image.resize(image, (RESOLUTION, RESOLUTION))
```

Comment:
nit: kerascv might already have a resizing layer which you could add to the augmenter

Reply:
But we cannot batch images together if they are of different sizes. So, this initial resizing step is necessary.

Reply:
What do you mean? That layer is specifically used for resizing to the same shape, it's used before dataset.batch

Reply:
`augmenter` is used after the batches have been formed. To form the initial batches, the images (to be present in the batches) need to be of the same shape. I know that the KerasCV resizing layer can be used for that purpose. But you mentioned adding the resizing layer to the `augmenter`. 

What am I missing?